File: group-similar-titles.py
def groupTitles(titles):
    groupedTitles = {}
    for title in titles:
        wordMap = [0]*26
        for c in title:
            index = ord(c) - ord("a")
            wordMap[index] += 1
        titleKey = "#".join(map(str, wordMap))
        if titleKey in groupedTitles:
            groupedTitles[titleKey].append(title)
        else:
            groupedTitles[titleKey] = [title]
    return groupedTitles.values()


titles = ["duel", "dule", "speed", "spede", "deul", "cars"]
query = "spede"

for groupedTitles in groupTitles(titles):
    if(query in groupedTitles):
        print(groupedTitles)

File: group-similar-titles.js
const alphabet = "abcdefghijklmnopqrstuvwxyz";

const groupTitles = (titles) => {
  const groupedTitles = new Map();
  titles.forEach((title) => {
    const wordMap = new Array(26).fill(0);
    [...title].forEach((letter) => {
      const letterIndex = alphabet.indexOf(letter);
      if (letterIndex >= 0) {
        wordMap[letterIndex]++;
      }
    });

    const titleKey = "#" + wordMap.join("#");
    const titlesByKey = groupedTitles.get(titleKey);
    const newTitles = titlesByKey ? [...titlesByKey, title] : [title];

    groupedTitles.set(titleKey, newTitles);
  });

  return groupedTitles.values();
};

const titles = ["duel", "dule", "speed", "spede", "deul", "cars"];
const query = "spede";
for (const groupedTitles of groupTitles(titles)) {
  if (groupedTitles.includes(query)) {
    console.log(groupedTitles);
  }
}
